src/appleseed.shaders/src/maya/as_maya_vectorProduct.osl

```diff
@@ -0,0 +1,113 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2016 Luis Barrancos, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Ref: $MAYA_LOCATION/docs/Nodes/vectorProduct.html
+
+shader as_maya_vectorProduct
+[[
+    string maya_node_name = "vectorProduct",
+    string maya_classification = "utility/general"
+]]
+(
+    // components = input1X|Y|Z
+    vector in_input1 = vector(0)
+    [[
+        string maya_attribute_name = "input1",
+        string maya_attribute_type = "vector",
+        string label = "Input Vector 1"
+    ]],
+    vector in_input2 = vector(0)
+    [[
+        string maya_attribute_name = "input2",
+        string maya_attribute_type = "vector",
+        string label = "Input Vector 2"
+    ]],
+    matrix in_matrix = matrix(1)
+    [[
+        string maya_attribute_name = "matrix",
+        string maya_attribute_type = "matrix",
+        string label = "Input Matrix"
+    ]],
+    int in_normalizeOutput = 0
+    [[
+        string maya_attribute_name = "normalizeOutput",
+        string maya_attribute_type = "bool",
+        string label = "Normalize Output",
+        string widget = "checkBox"
+    ]],
+    int in_operation = 1
+    [[
+        string maya_attribute_name = "operation",
+        string maya_attribute_type = "enum",
+        string label = "Operation",
+        string widget = "mapper",
+        string options = "None:0|Dot Product:1|Cross Product:2|Vector Matrix Product:3|Point Matrix Product:4"
+    ]],
+
+    // Components = outputX|Y|Z
+    output vector out_output = vector(0)
+    [[
+        string maya_attribute_name = "output",
+        string maya_attribute_type = "vector",
+        string label = "Output",
+        string widget = "null"
+    ]]
+)
+{
+    if ( in_operation == 0 )
```

Comment:
Our formatting convention doesn't put spaces before/after parentheses:

    if (in_operation == 0)

You can fix this in subsequent commits, as you go. No hurry.